src/memray/_memray.pyx

```diff
@@ -493,7 +492,8 @@ cdef class FileReader:
         except OSError as exc:
             raise OSError(f"Could not open file {file_name}: {exc.strerror}") from None
 
-        self._path = "/proc/self/fd/" + str(self._file.fileno())
+        # self._path = "/proc/self/fd/" + str(self._file.fileno())
```

Comment:
We open the file twice. This is an attempt to make sure that we get the same file both times, by opening it once and then re-opening a file handle to it later. We can try refactoring around this - ultimately the problem is that the `source` code is expecting a string for the file to open, and we want to pass it an already-opened FD, so that when we do our two passes over the file, they're both guaranteed to be over the _same_ file.

OTOH that was just me being paranoid, and perhaps that's not even a scenario we should bother worrying about - if the user replaces the file we're reading in the middle of a call to a reporter, maybe weird behavior is acceptable.

Reply:
Ok, I have preserved the Linux behaviour and leave it like `self._path = str(file_name)` for macos.